src/lib/components/search-attributes-form/search-attribute-row.svelte

```diff
@@ -32,44 +33,52 @@
     disableTypeForExisting = false,
     initialAttributeNames,
     onRemove,
+    onNameChange,
+    onTypeChange,
   }: Props = $props();
 
   const isTypeDisabled = $derived(
     submitting ||
-      (disableTypeForExisting &&
-        attribute.name &&
-        initialAttributeNames.has(attribute.name)),
+      (disableTypeForExisting && name && initialAttributeNames.has(name)),
   );
 
   const hasError = $derived(!!error);
+
+  const inputProps = $derived({
+    id: `attribute-name-${index}`,
+    value: name,
+    label: translate('search-attributes.attribute-label', {
+      index: index + 1,
+    }),
+    labelHidden: true,
+    disabled: submitting,
+    error: hasError,
+    oninput: (e: Event) =>
```

Comment:
nit: I believe you can use the `FormEventHandler` type to avoid coercion here, eg. `(e: FormEventHandler<HTMLInputElement>)`